ballerina/annotation_processor.bal

```diff
@@ -80,3 +82,31 @@ isolated function getIntrospection(GraphqlServiceConfig? serviceConfig) returns
     }
     return true;
 }
+
+isolated function getFieldInterceptors(service object {} serviceObj, string fieldName, string[] resourcePath) returns
+    readonly & (readonly & Interceptor)[] {
+    GraphqlResourceConfig? resourceConfig = getResourceConfig(serviceObj, fieldName, resourcePath);
+    if resourceConfig is GraphqlResourceConfig {
+        readonly & ((readonly & Interceptor)|(readonly & Interceptor)[]) interceptors = resourceConfig.interceptors;
+        if interceptors is (readonly & Interceptor) {
+            return [interceptors];
+        } else {
+            return interceptors;
+        }
+    }
+    return [];
+}
+
+isolated function getResourceConfig(service object {} serviceObj, string methodName, string[] resourcePath)
+returns GraphqlResourceConfig? {
+    any resourceAnnotation = getResourceAnnotation(serviceObj, resourcePath, methodName);
+    if resourceAnnotation is GraphqlResourceConfig {
+        return resourceAnnotation;
+    }
+    return;
+}
+
+isolated function getResourceAnnotation(service object {} serviceObject, string[] path, string methodName)
+returns any = @java:Method {
+    'class: "io.ballerina.stdlib.graphql.runtime.engine.Engine"
+} external;
```

Comment:
We can use a dependently-typed method here. In that case, we can omit the intermediate `getResourceConfig` method.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1316/commits/6346096ae72baf454266869638611f2fb109e779